### PR TITLE
Enable CAS for Add-ons

### PR DESCRIPTION
**Pull request by pvizeli**
This partial reverts commit 435241bccf3f3c678161b50cfefd4b036f1daba9.

## Proposed change

Using CAS for add-ons first

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/resolution/checks/core_trust.py [hunk 1]
```diff
@@ -32,8 +32,6 @@ async def run_check(self) -> None:
         except CodeNotaryUntrusted:
             self.sys_resolution.unhealthy = UnhealthyReason.UNTRUSTED
             self.sys_resolution.create_issue(IssueType.TRUST, ContextType.CORE)
-        except CodeNotaryBackendError:
-            _LOGGER.warning("CAS backend issue, skipping check")
         except CodeNotaryError:
             pass
```
[the remark this answers is not in the document]
Comment (pvizeli) [reply supervisor/resolution/checks/core_trust.py:36]: I don't have an opinion on that, but we can general send errors on sys_security to sentry if you would like it

**Review by mdegat01 — CHANGES_REQUESTED** [review]
Looks like there's some commented out code. Good to go otherwise.

File: supervisor/resolution/checks/core_trust.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdegat01) [reply supervisor/resolution/checks/core_trust.py:36]: Yea I feel like that would help us keep ahead of this. I can do it as a separate PR though.

**Comment by pvizeli** [comment]
> Looks like there's some commented out code. Good to go otherwise.

yeah, we going step by step

**Review by mdegat01 — APPROVED** [review]

**Pull request merged**